clang/include/clang/Analysis/Analyses/LifetimeSafety/LifetimeSafety.h

```diff
@@ -82,6 +89,15 @@ class LifetimeSafetyAnalysis {
   std::unique_ptr<LoanPropagationAnalysis> LoanPropagation;
 };
 } // namespace internal
+
+// utility function to print missing origin stats.
+void PrintStats(const LifetimeSafetyStats &Stats);
+
+/// The main entry point for the analysis.
+void runLifetimeSafetyAnalysis(AnalysisDeclContext &AC,
+                               LifetimeSafetyReporter *Reporter,
+                               LifetimeSafetyStats &Stats, bool CollectStats);
+
```

Comment:
please keep the public/non-internal functions to the top and before the internal namespace.